src/lcm/model.py

```diff
@@ -0,0 +1,120 @@
+"""Collection of classes that are used by the user to define the model and grids."""
+
+import dataclasses as dc
+from collections.abc import Callable
+from dataclasses import KW_ONLY, InitVar, dataclass, field
+
+from lcm.exceptions import ModelInitilizationError, format_messages
+from lcm.grids import Grid
+
+
+@dataclass(frozen=True)
+class Model:
+    """A user model which can be processed into an internal model.
+
+    Attributes:
+        description: Description of the model.
+        n_periods: Number of periods in the model.
+        functions: Dictionary of user provided functions that define the functional
+            relationships between model variables. It must include at least a function
+            called 'utility'.
+        choices: Dictionary of user provided choices.
+        states: Dictionary of user provided states.
+
+    """
+
+    description: str | None = None
+    _: KW_ONLY
+    n_periods: int
+    functions: dict[str, Callable] = field(default_factory=dict)
+    choices: dict[str, Grid] = field(default_factory=dict)
+    states: dict[str, Grid] = field(default_factory=dict)
+    _skip_checks: InitVar[bool] = False
```

Comment:
In which situation would you want to skip checks?

Reply:
Write custom testing model